Gems/MotionMatching/Code/Source/KdTree.cpp

```diff
@@ -83,9 +83,29 @@ namespace EMotionFX::MotionMatching
         m_maxDepth = maxDepth;
         m_minFramesPerLeaf = minFramesPerLeaf;
 
+        // Not all features are present in the KD-tree, thus we need to remap KD-tree local feature columns to the
+        // feature schema global feature columns.
+        AZStd::vector<size_t> localToSchemaFeatureColumns;
+        {
+            localToSchemaFeatureColumns.resize(m_numDimensions);
+
+            size_t currentColumn = 0;
+            for (const Feature* feature : features)
+            {
+                const size_t numDimensions = feature->GetNumDimensions();
+                const size_t featureColumnOffset = feature->GetColumnOffset();
+                for (size_t i = 0; i < numDimensions; ++i)
+                {
+                    localToSchemaFeatureColumns[currentColumn] = featureColumnOffset + i;
+                    currentColumn++;
+                }
+            }
+
+            AZ_Assert(m_numDimensions == currentColumn, "There should be a column index mapping for each of the available dimensions.");
+        }
+
         // Build the tree.
-        m_featureValues.resize(m_numDimensions);
-        BuildTreeNodes(frameDatabase, featureMatrix, features, new Node(), nullptr, 0);
+        BuildTreeNodes(frameDatabase, featureMatrix, localToSchemaFeatureColumns, new Node(), nullptr, 0);
```

Comment:
nit/question: Should we be using `aznew` here? Or maybe even update this to `make_unique` to better handle the memory management?

Reply:
Added the allocator decl to the classes and changed to aznew